pkg/controller/node/rate_limited_queue_test.go

```diff
@@ -152,25 +152,58 @@ func TestTry(t *testing.T) {
 }
 
 func TestTryOrdering(t *testing.T) {
+	defer func() { now = time.Now }()
+	current := time.Unix(0, 0)
+	delay := 0
+	// the current time is incremented by 1ms every time now is invoked
+	now = func() time.Time {
+		if delay > 0 {
```

Comment:
Single threaded, Try is a for {} with a rate limiter and a check.

On Mon, Apr 18, 2016 at 1:42 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/controller/node/rate_limited_queue_test.go
> <https://github.com/kubernetes/kubernetes/pull/24382#discussion_r60101564>
> :
>
> > @@ -152,25 +152,58 @@ func TestTry(t *testing.T) {
> >  }
> >
> >  func TestTryOrdering(t *testing.T) {
> > +	defer func() { now = time.Now }()
> > +	current := time.Unix(0, 0)
> > +	delay := 0
> > +	// the current time is incremented by 1ms every time now is invoked
> > +	now = func() time.Time {
> > +		if delay > 0 {
>
> does this need a lock, or is now() only called from a single goroutine?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24382/files/7245730ea5a1ecc63e0e97cc617aec2e25b4921a#r60101564>
>